### DISCUSSION TITLE
Google Analytics using gtag plugin not showing page titles

### DISCUSSION THREAD

**Discussion by oripka** [nuxt/nuxt#17061] [Questions]
As others I ran into this issue: https://github.com/MatteoGabriele/vue-gtag/issues/346. The author of the plugin mentioned a bug report in the Nuxt repository, which I was not able to find.

I am using this plugin code `plugins/vue-gtag.client.js`

```
import VueGtag from 'vue-gtag'

/* https://stackoverflow.com/questions/71540309/how-to-connect-google-analytics-to-nuxt3-app */
export default defineNuxtPlugin((nuxtApp) => {
    nuxtApp.vueApp.use(VueGtag, {
        config: {
            id: 'G-XXXXX',
        },
    }, nuxtApp.$router)
})
```
I also tried this code `plugins/vue-gtag-next.client.js`

```
import VueGtag from 'vue-gtag-next'

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.use(VueGtag, {
    property: {
      id: 'G-XXXXX'
    }
  }, nuxtApp.$router)
})
```

As mentioned in the bug report the title (or `dt` parameter sent to `https://regionXXX.google-analytics.com/g/collect` is not correct. The page title isn't always logged and instead sometimes falls back to the page's path '/page-title'.

Also in the Nuxt3 docs about [Vue plugins](https://v3.nuxtjs.org/guide/directory-structure/plugins/#vue-plugins) it doesn't mentioned that one needs to pass `nuxtApp.$router`, which I found necessary in both version.

How do people get good data into google analytics?

**Comment by mirumirumi** [comment]
Hi!
I am also using vue-gtag and the following code is getting each page title without any problem. Have you already tried it?

analytics.client.ts:
```ts
import { defineNuxtPlugin } from "#app"
import VueGtag from "vue-gtag"

export default defineNuxtPlugin(nuxtApp => {
  const router = useRouter()

  nuxtApp.vueApp.use(VueGtag, {
    appName: "My Great Site",
    pageTrackerScreenviewEnabled: true,
    config: { id: "G-xxxxxxxxxx" },
  }, router)
})
```

**Comment by sxgrant** [comment]
gtag seems to have changed their initialisation at some point.  This is what worked for me

```
import VueGtag, { trackRouter } from 'vue-gtag-next'

export default defineNuxtPlugin((nuxtApp) => {
    const config = useRuntimeConfig()
    const router = useRouter()

    nuxtApp.vueApp.use(VueGtag, {
        property: { id: config.public.googleAnalyticsId },
    })

    trackRouter(router)
})
```
